src/ui/components/SearchCoordinates/component.ts

```diff
@@ -0,0 +1,32 @@
+import Component, { tracked } from '@glimmer/component';
+
+export default class SearchForm extends Component {
+  args: {
+    onSearch: (coordinates: Promise<number[]>) => void;
+  };
+
+  searchByLocation(event) {
+    if (event) {
+      event.preventDefault();
+    }
+
+    let { onSearch } = this.args;
+
+    let coordinatesPromise: Promise<number[]> = new Promise((resolve, reject) => {
+      let onSuccess = (position) => {
+        let { latitude, longitude } = position.coords;
+        resolve([latitude, longitude]);
+      };
+
+      let onError = (e) => {
+        reject(e.code);
```

Comment:
I'd just reject with the full error here. It's up to the caller of this to decide what attributes of the error it cares about.